src/ensembl/src/content/app/entity-viewer/state/gene-view/view/geneViewViewSlice.ts

```diff
@@ -144,19 +118,93 @@ GeneViewTabMap.set(View.GENE_PANELS, {
   secondaryTab: GeneRelationshipsTabName.GENE_PANELS
 });
 
-export const defaultEntityViewerGeneViewUIState: EntityViewerGeneViewUIState = {
-  view: View.TRANSCRIPTS,
+export type SelectedTabViews = Record<
+  'geneFunctionTab' | 'geneRelationshipsTab',
+  View | null
+>;
+
+export type ViewStatePerGene = {
+  current: View;
```

Comment:
Why do you prefer `current` here instead of just `view`?

Reply:
There are already so many `view`s in this model :smile: 

I named the field that holds this slice in `entityViewerGeneViewReducer` `view`:

```
export default combineReducers({
  view: geneViewViewReducer,
  transcripts: geneViewTranscriptsReducer,
  proteins: geneViewProteinsReducer
});
```

So when faced with the prospect of having this structure: `state.entityViewer.geneView.view[genomeId][geneId].view`, I thought that `state.entityViewer.geneView.view[genomeId][geneId].current` will be slightly better.

As you can see, I was unable to come up with a good name for what's now called `geneViewViewReducer` :disappointed:

Reply:
geneViewGeneReducer? That would be more consistent with the other names. Unless I've missed something and you're already using this name somewhere else.

Reply:
But this particular reducer isn't responsible for gene at all. It holds a very small piece of state that is currently responsible for which tab to show to the user: whether it's Transcripts, Gene Function or GeneRelationships, and whether within GeneFunction it's Proteins, Variants, Phenotypes, etc.

It's almost as if it is `geneViewTabsReducer`, but I was hesitant to call it this way. `geneViewUIReducer` would be too broad, because the `transcripts` and the `proteins` slices of this state are also responsible for the UI — they keep track of which transcripts/proteins are expanded or collapsed.